numba_cuda/numba/cuda/tests/nrt/test_nrt.py

```diff
@@ -76,6 +149,76 @@ def g(out_ary):
 
         self.assertEqual(out_ary[0], 1)
 
+    def test_nrt_detect_linked_ptx_file(self):
+        src = f"#include <{get_include()}/nrt.cuh>"
+        src += """;
+                 extern "C" __device__ int extern_func(int* nb_retval){
+                     NRT_Allocate(1);
```

Comment:
Hmm, this test case leaks memory right? Even though it's 1 byte it may not pass the memory leak tests.

Makes me feel that we should also make a binding for `NRT_Deallocate` and use that following `allocate_shim`.